dcos/data_source_dcos_service_account_secret.go

```diff
@@ -0,0 +1,146 @@
+package dcos
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type ServiceAccontJson struct {
+	Scheme        string `json:"scheme"`
+	UID           string `json:"uid"`
+	LoginEndpoint string `json:"login_endpoint"`
+	PrivateKey    string `json:"private_key"`
+}
+
+func dataSourceDcosServiceAccountSecret() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDcosServiceAccountSecretRead,
+		Schema: map[string]*schema.Schema{
+			"private_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The PEM encoded PKCS1 (RSA) or PKCS8 private key for this account",
+			},
+			"uid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The user ID",
+			},
+			"login_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     "https://leader.mesos/acs/api/v1/auth/login",
+				Description: "Customize the login_endpoint parameter",
+			},
+			"contents": {
```

Comment:
what about naming this `"json"` ?

Reply:
I was thinking that this should be treated as an opaque variable, and hence the content-agnostic `content` variable.

The user shouldn't care if the contents are JSON or anything that we might consider in the future 🤔

Reply:
ok that also makes sense to me